server/jetstream_cluster.go

```diff
@@ -7245,7 +7245,7 @@ func decodeStreamMsg(buf []byte) (subject, reply string, hdr, msg []byte, lseq u
 	if len(buf) < rl {
 		return _EMPTY_, _EMPTY_, nil, nil, 0, 0, errBadStreamMsg
 	}
-	reply = string(buf[:rl])
+	reply = bytesToString(buf[:rl])
```

Comment:
This is going to be used, for instance in `applyStreamEntries()`, to send a flow control message, which means pushing a new object that will reference this string into an IPQ that will use that from a different go routine. I don't think it is safe, right?

Reply:
I think this one is ok.

Reply:
The buffer belongs to a `CommittedEntry` (when called from `applyStreamEntries`), which is then returned to a pool in `monitorStream`. Are we certain that the underlying buffer is never reused, or is not a reference to a read/shared buffer of some sort?
For sure, if the buffer can be modified, and since the "reply" subject is referenced and used in a different go routine (follow the path of sendFlowControlReply), then it could end-up being wrong (see https://go.dev/play/p/IqRCEp2NEmA, where subject ends up being `ZBC` (instead of `ABC`)).
If the buffer is not reused, what I don't know is if Golang - when using `unsafe.String()` actually retains the underlying buffer or not.

Reply:
Will revert.